contracts/token/ERC721/extensions/ERC721Consecutive.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.1 (token/ERC721/extensions/ERC721Burnable.sol)
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "../../../utils/Checkpoints.sol";
+import "../../../utils/math/SafeCast.sol";
+import "../../../utils/structs/BitMaps.sol";
+
+/**
+ * @title ERC721 Cheap sequential minting
+ */
+abstract contract ERC721Consecutive is ERC721 {
+    using BitMaps for BitMaps.BitMap;
+    using Checkpoints for Checkpoints.Checkpoint160[];
+
+    Checkpoints.Checkpoint160[] private _sequentialOwnership;
+    BitMaps.BitMap private _sequentialBurn;
+
+    event ConsecutiveTransfer(
```

Comment:
Do we want to put that in an IERC2309 ? Its final

Reply:
Yes I'd say so. We should have that interface.